src/amp-story-player/amp-story-player-impl.js

```diff
@@ -565,6 +569,21 @@ export class AmpStoryPlayer {
     this.isLaidOut_ = true;
   }
 
+  /**
+   * Returns a promise that resolves when story in given iframe is finished
+   * loading.
+   * @param {number} iframeIdx
+   * @return {!Promise}
+   * @private
+   */
+  waitForStoryToLoadPromise_(iframeIdx) {
+    return new Promise((resolve) => {
```

Comment:
Did you decide to not implement a timeout because of the fallback you have where if the user navigates to the next story it still loads properly?